### PR TITLE
Deprecate passing args as positional in sort_values

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] xref #41485
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -647,6 +647,7 @@ Deprecations
 - Deprecated setting :attr:`Categorical._codes`, create a new :class:`Categorical` with the desired codes instead (:issue:`40606`)
 - Deprecated behavior of :meth:`DatetimeIndex.union` with mixed timezones; in a future version both will be cast to UTC instead of object dtype (:issue:`39328`)
 - Deprecated using ``usecols`` with out of bounds indices for ``read_csv`` with ``engine="c"`` (:issue:`25623`)
+- Deprecated passing arguments as positional (except for ``by``) in :meth:`DataFrame.sort_values` and :meth:`Series.sort_values` (:issue:`41485`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.3.0.rst:650]: this change won't solve the `error: Signature of "sort_values" incompatible with supertype "NDFrame"  [override]`. maybe we could also give `by` a default value (the dataframe columns). The 'by' keyword could still be passed positionally if it has a default value.

Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.3.0.rst:650]: we could also perhaps deprecate the axis keyword on Series at the same time to clean this up further.

Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.3.0.rst:650]:
```suggestion
- Deprecated passing arguments as positional in :meth:`DataFrame.sort_values` (except for ``by``) and :meth:`Series.sort_values` (:issue:`41485`)
```

File: pandas/tests/frame/methods/test_sort_values.py [hunk 2]
```diff
@@ -856,3 +856,13 @@ def test_sort_column_level_and_index_label(
...
+
+    def test_sort_values_pos_args_deprecation(self):
+        # https://github.com/pandas-dev/pandas/issues/41485
+        df = DataFrame({"a": [1, 2, 3]})
+        msg = (
+            r"Starting with Pandas version 2\.0 all arguments of sort_values except "
+            r"for the arguments 'self' and 'by' will be keyword-only"
+        )
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            df.sort_values("a", 0)
```
Comment (jorisvandenbossche) [inline pandas/tests/frame/methods/test_sort_values.py:868]: Same comment as in other PRs regarding asserting the result

File: doc/source/whatsnew/v1.3.0.rst (hunk 1 above)
Comment (MarcoGorelli) [reply doc/source/whatsnew/v1.3.0.rst:650]: > we could also perhaps deprecate the axis keyword on Series at the same time to clean this up further.

I don't understand, wouldn't this be incompatible with the supertype?

> maybe we could also give by a default value (the dataframe columns). The 'by' keyword could still be passed positionally if it has a default value.

Sure, could that - that could be done directly in 1.3 without a prior warning, right? As current behaviour would be preserved

Comment (jorisvandenbossche) [reply doc/source/whatsnew/v1.3.0.rst:650]: > maybe we could also give by a default value (the dataframe columns). The 'by' keyword could still be passed positionally if it has a default value.

That might be a good idea, but I would personally leave that for a separate PR (as it seems an independent new feature)

**Pull request merged**